### DISCUSSION TITLE
Pico W disconnection

### DISCUSSION THREAD

**Discussion by ebolisa** [micropython/micropython#11113] [General]
Hi, I posted my issue on a FB group but decided this’s a better place to post it. So, here it is…

Hey, is anyone having internet connections stability issues with the pico w?
I’m going nuts with an alarm clock project and I cannot find what’s causing the board loose connections. I know the code is good as it’s been running for months on ESPs. 

What the code does:
- Gets credentials from a json file and connects to internet. 
- Gets epoch time from a site and updates the local time every 3 hrs (using my own settime). 
- Gets the board’s temperature and local temperature from a weather station site every 5 minutes. 
- Runs a web server to set the alarm and other settings. 

What I’ve tried so far with no avail. 
- Timers
- Non blocking timers
- Asyncio
- Gc.collect() in the main loop. 
- What I’ve not tried yet is the wathdog timer. 

The problem:
It’s not consistent. The time and temperatures are displayed as designed then, after a x period of time, it hangs up with the display showing the last hour/minutes. I noticed it happens when urequests is used to retrieve the json data. The line if wlan.isconnected() is used prior to that but the try/except still catches a no internet issue with error -1 or -2. 

Having exposed the above, have any of you experienced this type of issue? I know the actual OS is not stable but, I don’t think it’s causing the problem or, maybe it is 🤷🏻‍♂️

**Comment by peterhinch** [comment]
The OS, though not yet an official release, is stable. The issue may be [as described here](https://github.com/peterhinch/micropython-samples/blob/master/resilient/README.md). The Pico W has proved reliable running [this library](https://github.com/peterhinch/micropython-mqtt/tree/master/mqtt_as) which is designed for resilience.
